test/unit/services/finance/stripe_charge_service_test.rb

```diff
@@ -105,6 +105,14 @@ class Finance::StripeChargeServiceTest < ActiveSupport::TestCase
     assert_equal 'requires_payment_method', payment_intent.reload.state
   end
 
+  test 'sets payment intent description' do
+    response = build_response(true, 'Transaction Approved', object: 'payment_intent', id: 'new-payment-intent-id', status: 'succeeded')
+    invoice.send(:issue!)
+    gateway.expects(:purchase).with(anything, anything, has_entry(:description, "#{invoice.provider.name} API services fix")).returns(response)
+
+    assert service.send(:create_payment_intent, amount)
```

Comment:
There are two "send" in this test. Are those methods `issue!` (sounds like state machine) and `create_payment_intent` private methods by any chance?
If so, should we be testing those (more like the `create_payment_intent` one, as it's in the assertion) instead of the public methods that call them?

Reply:
Yes, `issue!` is private (it is a state machine indeed), and `create_payment_intent` is protected.

In this case we're just testing what the payment gateway receives, and this is just the easiest way to prepare the invoice for test and triggering the charging event.
But I can change it to use public methods, I think should be easy.

Reply:
Not a real issue for me, as long as we're not skipping code that could have an impact on the test, which's what `send` makes me suspicious of 😅.
Oh... And I just saw you already changed it.
I'm resolving this one :v: